lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -1168,34 +1168,58 @@ bool swift::diagnoseNonSendableTypesInReference(
       return true;
   }
 
-  // For functions, check the parameter and result types.
+  // For functions or subscripts, check the parameter and result types.
   SubstitutionMap subs = declRef.getSubstitutions();
-  if (auto function = dyn_cast<AbstractFunctionDecl>(declRef.getDecl())) {
+  auto decl = declRef.getDecl();
+  if (isa<AbstractFunctionDecl>(decl) || isa<SubscriptDecl>(decl)) {
     if (funcCheckOptions.contains(FunctionCheckKind::Params)) {
       // only check params if funcCheckKind specifies so
-      for (auto param : *function->getParameters()) {
+      ParameterList *paramList = nullptr;
+      if (auto function = dyn_cast<AbstractFunctionDecl>(decl)) {
+        paramList = function->getParameters();
+      } else if (auto subscript = dyn_cast<SubscriptDecl>(decl)) {
+        paramList = subscript->getIndices();
+      }
+
+      // Check params of this function or subscript override for sendability
+      for (auto param : *paramList) {
         Type paramType = param->getInterfaceType().subst(subs);
+        if (param->isSending() && !paramType->hasError()) {
```

Comment:
Should this check instead be a part of `diagnoseNonSendableTypes`? We do pass a declaration there, it could be declaration of the parameter instead of the whole function. This might make it easier to diagnose result types properly as well.

Reply:
Thank you for checking.

I moved the check into a new method `diagnoseNonSendableTypesWithSendingCheck`. https://github.com/swiftlang/swift/pull/78601/commits/765f8acf7bd13df51db6887721f395aff8ef4b87

How about this?

Reply:
Yes, that looks great, thank you!